FastExcel/CellRange.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace FastExcel
+{
+    /// <summary>
+    /// Represents a range of cells
+    /// </summary>
+    public class CellRange
+    {
+        internal string SheetName;
+        public string ColumnStart, ColumnEnd;
+        public int RowStart = 1;
+        public int? RowEnd;
+
+        /// <summary>
+        /// Defines a range of cells using a reference string
+        /// </summary>
+        /// <param name="reference">Reference string i.e. Sheet1!$A$1</param>
+        internal CellRange(string reference)
+        {
+            string[] splitReference = reference.Split('!');
```

Comment:
Yeah, I think the regex check should be more complex in checking for a valid reference. This way if there happens to be an invalid reference, there is no null reference errors.
I checked and in excel you can type anything into the reference field, so we do need to check thoroughly.